packages/react/src/side-nav/SideNav.test.tsx

```diff
@@ -55,4 +54,276 @@ describe('SideNav', () => {
 			expect(el).toHaveAccessibleName(defaultTestingProps.title);
 		});
 	});
+
+	describe('when showSubLevel is "whenActive"', () => {
+		describe('when there is a mix of zero and more sub-level items', () => {
+			describe('when the active item has no sub-level items', () => {
+				beforeEach(() => {
+					render(
+						<SideNav
+							{...defaultTestingProps}
+							activePath="/welcome"
+							showSubLevel="whenActive"
+						/>
+					);
+
+					const user = userEvent.setup();
+
+					user.click(screen.getByRole('button', { name: 'In this section' }));
+				});
+
+				test('then no sub-level items should be visible', async () => {
+					const levelOneItemHrefs = defaultTestingProps.items.map(
+						({ href }) => href
+					);
+
+					const navItems: HTMLAnchorElement[] = await screen.findAllByRole(
+						'link'
+					);
+					const navItemPathnames = navItems
+						.map((item) => new URL(item.href).pathname)
+						.filter((href) => href !== '/'); // Remove the heading's link
+
+					expect(navItemPathnames).toEqual(levelOneItemHrefs);
+				});
+
+				test('then icons indicating hidden sub-level items should be visible', async () => {
+					expect(
+						await screen.findAllByRole('img', {
+							name: /. Has [0-9] sub-level links./,
```

Comment:
May need changing with the pluralising, but SonarCloud whinges which I kinda agree with…
```suggestion
							name: /. Has \d sub-level links./,
```

Reply:
Ugh SonarCloud. I can make this change but it's such a pointless waste of time, IMO. Arguably, it's harder to read.

Reply:
Actually it won't work cleanly with pluralisation anyway.